test/e2e/common/node/podtemplates.go

```diff
@@ -159,6 +161,44 @@ var _ = SIGDescribe("PodTemplates", func() {
 
 		err = wait.PollImmediate(podTemplateRetryPeriod, podTemplateRetryTimeout, checkPodTemplateListQuantity(f, "podtemplate-set=true", 0))
 		framework.ExpectNoError(err, "failed to count required pod templates")
+
+	})
+
+	ginkgo.It("should replace a pod template", func() {
+		ptClient := f.ClientSet.CoreV1().PodTemplates(f.Namespace.Name)
+		ptName := "podtemplate-" + utilrand.String(5)
+
+		ginkgo.By("Create a pod template")
+		ptResource, err := ptClient.Create(context.TODO(), &v1.PodTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: ptName,
+			},
+			Template: v1.PodTemplateSpec{
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{Name: "e2e-test", Image: imageutils.GetE2EImage(imageutils.Agnhost)},
+					},
+				},
+			},
+		}, metav1.CreateOptions{})
+		framework.ExpectNoError(err, "failed to create pod template")
+
+		ginkgo.By("Replace a pod template")
+		ptResource.Annotations = map[string]string{
+			"updated": "true",
+		}
+		updatedPT, err := ptClient.Update(context.TODO(), ptResource, metav1.UpdateOptions{})
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			ptResource, err = ptClient.Get(context.TODO(), ptName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "Unable to get pod template %s", ptName)
```

Comment:
ptResource.Annotations will not contain `updated=true` annotation after the `Get`.